frontend/src/__tests__/cypress/cypress/pages/pipelines/restoreRunModal.ts

```diff
@@ -0,0 +1,21 @@
+import { Modal } from '~/__tests__/cypress/cypress/pages/components/Modal';
+
+class RestoreRunModal extends Modal {
+  protected testId;
+
+  constructor(testId = 'restore-run-modal') {
+    super('Restore run?');
+    this.testId = testId;
+  }
+
+  find() {
+    return cy.findByTestId(this.testId).parents('div[role="dialog"]');
```

Comment:
```suggestion
    return this.find().findByTestId(this.testId).parents('div[role="dialog"]');
```
same suggestion as above

Reply:
This function is meant to replace the "find" of Modal for this instance/sub-class of Modal, so I don't think it makes sense here.